forgerock-openbanking-uk-aspsp-common/src/test/java/com/forgerock/openbanking/common/services/frequency/FrequencyServiceTest.java

```diff
@@ -0,0 +1,126 @@
+import org.joda.time.DateTime;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.catchThrowableOfType;
+
+/**
+ * Unit test for {@link FrequencyService}.
+ */
+@RunWith(Parameterized.class)
+public class FrequencyServiceTest {
+
+    private final String frequencyInterval;
+    private final String wrongFrequencyInterval;
+    private final DateTime firstDateTime;
+    private final FrequencyService frequencyService;
+    private static final String WRONG = "WRONG";
+
+    public FrequencyServiceTest(String frequencyInterval, String wrongFrequencyInterval, DateTime firstDateTime) {
+        this.frequencyInterval = frequencyInterval;
+        this.wrongFrequencyInterval = wrongFrequencyInterval;
+        this.firstDateTime = firstDateTime;
+        this.frequencyService = new FrequencyService();
+    }
+
+    @Parameterized.Parameters(
+            name = "{index}: frequencyInterval ({0}), firstDateTime ({2}), wrongFrequencyInterval ({1})"
+    )
+    public static Collection<Object[]> frequencies() {
+
+        List<Object[]> args = new ArrayList<>();
+        args.add(new Object[]{FrequencyType.EVERYDAY.getFrequencyStr(),
+                FrequencyType.EVERYDAY.getFrequencyStr() + WRONG
+                ,DateTime.now()});
+
+        args.add(new Object[]{FrequencyType.EVERYWORKINGDAY.getFrequencyStr(),
+                FrequencyType.EVERYWORKINGDAY.getFrequencyStr() + WRONG
+                ,DateTime.now()});
+
+        args.add(new Object[]{FrequencyType.INTERVALWEEKDAY.getFrequencyStr() + ":01:07",
+                FrequencyType.INTERVALWEEKDAY.getFrequencyStr() + ":01:08",
+                DateTime.now()});
+
+        args.add(new Object[]{FrequencyType.WEEKINMONTHDAY.getFrequencyStr() + ":02:03",
+                FrequencyType.WEEKINMONTHDAY.getFrequencyStr() + ":00:03",
+                DateTime.now()});
+
+        args.add(new Object[]{FrequencyType.INTERVALMONTHDAY.getFrequencyStr() + ":01:30",
+                FrequencyType.INTERVALMONTHDAY.getFrequencyStr() + ":1:-8",
+                DateTime.now()});
+
+        args.add(new Object[]{FrequencyType.QUARTERDAY.getFrequencyStr() + ":ENGLISH",
+                FrequencyType.QUARTERDAY.getFrequencyStr() + ":SENT",
+                DateTime.now()});
+
+        args.add(new Object[]{FrequencyType.INTERVALDAY.getFrequencyStr() + ":02",
+                FrequencyType.INTERVALDAY.getFrequencyStr() + ":01",
+                DateTime.now()});
+
+        return args;
+    }
+
+    @Test
+    public void shouldMatchPattern_frequencyInterval() {
+        // Given
+        // this.firsDateTime
+        // this.frequencyInterval
+
+        // When
+        DateTime dateTime = frequencyService.getNextDateTime(firstDateTime, frequencyInterval);
```

Comment:
To be honest, I wish I hadn't mentioned ParameterizedTests! 😆 (though I did say if we were using JUnit 5 rather than JUnit 4).

Unfortunately this conciseness comes at a cost. The values are now separated from the test and it's difficult to know exactly what's being tested and what the expected behaviour should be (and I still don't think `isNotNull()` is a very thorough assertion).

I know you understand it, but you have to put yourself in the eyes of someone who sees this for the first time and isn't familiar with the `FrequencyService` or the various intervals etc.

I appreciate you've tried to improve things, but I honestly think the previous version was easier to understand. Sorry! 😄

Reply:
Rollback done.